.ci/lib.sh

```diff
@@ -327,6 +327,54 @@ build_install_parallel() {
 	rm -rf "$gnu_parallel_dir" "$gnu_parallel_tar_pkg"
 }
 
+build_install_cmake() {
+	local version=$(get_version "externals.cmake.version")
+	local url=$(get_version "externals.cmake.url")
+	[ "${url}" != "null" ] && [ "${version}" != "null" ]
+	local file="cmake-${version}.tar.gz"
+	local dir="cmake-${version}"
+	local download="${url}/releases/download/v${version}/${file}"
+
+	pushd /root
+	wget "${download}"
+	tar -zxf "${file}"
+	cd "${dir}"
+	./bootstrap
+	make
+	make install
+	popd
+}
+
+build_install_musl() {
```

Comment:
This function is extremely similar to the one above. I'd do one of the following:

- Create a generic function that can download a source package and then make both these functions call it.
- Steal existing code from one of the other repos that does this.
  I'm *sure* we have/had a function to download build and install a generic source package in `tests/.ci/` or similar but cannot find it atm. /cc @chavafg, @grahamwhaley.

Reply:
It is not generic, we will have 
case $1 in
      musl)
;;
     camke)
;; 
in scripts to build download url. Since download url is quit different

Do you have some idea to make it simple?

Reply:
Let's defer that for a follow-up PR maybe.